### ISSUE TITLE
Sending string values vs. objects

### ISSUE THREAD

**Issue by gotmikhail** [firebase/firebase-admin-node#102]
Operating System version: CentOS
Firebase SDK version: 5.4.1
Firebase Product: Cloud Messaging

https://github.com/firebase/firebase-admin-node/blob/fc45c4f5da404e21934cbd9f5a4e7116cb744f0a/src/messaging/messaging.ts#L715

Is there any particular reason why we're restricting the values of the `key:value` pairs in `data` to just `string`s?

If calling `https://fcm.googleapis.com/fcm/send` directly, you can include `objects` within `data`.

**Comment by hiranya911** [comment]
The FCP server [API reference](https://firebase.google.com/docs/cloud-messaging/http-server-ref#downstream-http-messages-json) states:

```
Values in string types are recommended. You have to convert values in objects
or other non-string data types (e.g., integers or booleans) to string.
```

**Comment by hiranya911** [comment]
@shasd can you also please take a look?

**Comment by stychu** [comment]
yeah Im also curious aboyt this. Im facing the problem how to send my data payload from server to the client. Any suggestions on how to stringify/parse the payload automatically so i dont have to remember that data payload can be set as 
```
const payload = {
    data: {
        body: 1
    }
};
``` 
and it will be stringified or somthing.

rather than always remeber to make this as string ?

```
const payload = {
    data: {
        body: '1'
    }
};
```

**Comment by hiranya911** [comment]
Perhaps look into TypeScript? Our type definitions enforce this correctly: 

```
[key: string]: string;
```

**Comment by redward** [comment]
>  string;

Any changes ? how to add JSON object within data payload ?

**Comment by mohammad-kazemzadeh** [comment]
@redward 
You'll have to Stringify the object.
`
const payload = {
    data: {
        foo: JSON.Stringify({foo: "bar"}),
        body: "Notification Body",
        ...
    }
};
`
